iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/OpenHouseInternalCatalog.java

```diff
@@ -101,16 +96,16 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
           houseTableRepositoryException);
     }
     if (purge) {
-      // delete data and metadata files on hdfs
-      try {
-        FileSystem fs = fsStorageProvider.storageClient();
-        fs.delete(new Path(tableLocation), true);
-      } catch (IOException e) {
-        throw new UncheckedIOException(
-            String.format(
-                "Deleting Directory Failed databaseId: %s, tableId: %s, tableLocation: %s",
-                identifier.namespace().toString(), identifier.name(), tableLocation),
-            e);
+      // Delete data and metadata files from storage.
+      if (fileIO instanceof SupportsPrefixOperations) {
+        log.debug("Deleting files for table {}", tableLocation);
+        ((SupportsPrefixOperations) fileIO).deletePrefix(tableLocation);
+      } else {
+        log.debug(
+            "Failed to delete files for table {}. fileIO does not support prefix operations.",
+            tableLocation);
+        throw new UnsupportedOperationException(
+            "Drop table is supported only with a fileIO instance that SupportsPrefixOperations");
```

Comment:
is this temporary? What does this branch look like when we supports blobIO?

Reply:
This is not temporary. This moves us to a state where catalog no longer has any dependency on FileSystem interface. Having dependency on both FileIO and FileSystem adds complexity specially when they are using something other than HDFS.

This will allow us to support BlobFileIO seamlessly. All major FileIO plugins like S3FileIO extend from SupportsPrefixOperations. So, will our plugin for Blobstore.